src/transformers/models/mask2former/configuration_mask2former.py

```diff
@@ -0,0 +1,336 @@
+    [`Mask2FormerModel`]. It is used to instantiate an Mask2Former model according to the specified arguments, defining
+    the model architecture. Instantiating a configuration with the defaults will yield a similar configuration to that
+    of the Mask2Former
+    [facebook/mask2former-instance-swin-small-coco](https://huggingface.co/facebook/mask2former-instance-swin-small-coco)
+    architecture. Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs.
+    Read the documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        feature_size (`int`, *optional*, defaults to 256):
+            The features (channels) of the resulting feature maps.
+        mask_feature_size (`int`, *optional*, defaults to 256):
+            The masks' features size, this value will also be used to specify the Feature Pyramid Network features'
+            size.
+        hidden_dim (`int`, *optional*, defaults to 256):
+            Dimensionality of the encoder layers
+        encoder_feedforward_dim (`int`, *optional*, defaults to 1024):
+            Dimension of feedforward network for deformable detr encoder used as part of pixel decoder
+        norm (`str`, *optional*, defaults to `"GN"`):
+            Normalization for all convolution layers
+        encoder_layers (`int`, *optional*, defaults to 6):
+            Number of layers in the deformable detr encoder used as part of pixel decoder.
+        decoder_layers (`int`, *optional*, defaults to 10):
+            Number of layers in the Transformer decoder used by Mask2Former
+        num_heads (`int`, *optional*, defaults to 8):
+            Number of attention heads for each attention layer
+        dropout (`float`, *optional*, defaults to 0.1):
+            The dropout probability for all fully connected layers in the embeddings, encoder
+        dim_feedforward (`int`, *optional*, defaults to 2048):
+            Feature dimension in feedforward network for transformer decoder
+        pre_norm (`boolean``, *optional*, defaults to False):
+            Whether to use pre-LayerNorm or not in layers of transformer decoder
+        enforce_input_proj (`boolean``, *optional*, defaults to False):
+            add input project 1x1 conv even if input channels and hidden dim is identical in transformer decoder
+        common_stride (`int`, *optional*, defaults to 4):
+            parameter used for determining number of FPN levels used as part of pixel decoder
+
+    Example:
+    ```python
+    >>> from transformers import Mask2FormerDecoderConfig, Mask2FormerConfig, Mask2FormerModel
+
+    >>> # Initializing a Mask2FormerDecoderConfig with facebook/mask2former-instance-swin-small-coco style configuration
+    >>> decoder_config = Mask2FormerDecoderConfig()
+    >>> configuration = Mask2FormerConfig(decoder_config=decoder_config.to_dict())
```

Comment:
```suggestion
    >>> configuration = Mask2FormerConfig(decoder_config)
```
Would be nice if this works too